### PR TITLE
Setup Travis and add necessary PEP8 conformance tests

### LINKED ISSUE #7: Setup travis for hydra-python-core
Add a .travis.yml file to enable travis on this repo. The `.travis.yml` file should run checks for PEP8 compliance along with other tests.

### PR DESCRIPTION
**Pull request by de-sh**
Fixes #7 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

This PR adds a .travis.yml CI config file and a PEP8 test that is run with unittest framework.

### Change logs
- Travis CI configured
- PEP8 test added
- PEP8 conformance in all source files

### CODE REVIEW

File: hydra_python_core/doc_maker.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from typing import Any, Dict, Match, Optional, Tuple, Union
 
 
-def error_mapping(body: str=None) -> str:
+def error_mapping(body: str = None) -> str:
```
Comment (chrizandr) [inline hydra_python_core/doc_maker.py]: Is this PEP8 compliant?
https://stackoverflow.com/questions/8853063/pep-8-why-no-spaces-around-in-keyword-argument-or-a-default-parameter-value

File: hydra_python_core/doc_writer.py [hunk 2]
```diff
@@ -108,8 +110,8 @@ class HydraClass():
     """Template for a new class."""
 
     def __init__(
-            self, id_: str, title: str, desc: str, path: str=None,
-            endpoint: bool=False, sub_classof: None=None) -> None:
+            self, id_: str, title: str, desc: str, path: str = None,
+            endpoint: bool = False, sub_classof: None = None) -> None:
```
Comment (chrizandr) [inline hydra_python_core/doc_writer.py]: Same here.

File: hydra_python_core/doc_maker.py (hunk 1 above)
Comment (de-sh) [reply hydra_python_core/doc_maker.py]: Ok, I didn't see that, but it passes the test. Also I used AutoPEP8 to get this output.

**Comment by de-sh** [comment]
@chrizandr necessary changes made :+1:

**Pull request merged**